packages/flutter_tools/lib/src/resident_runner.dart

```diff
@@ -737,6 +739,35 @@ abstract class ResidentRunner {
     throw Exception('Canvaskit not supported by this runner.');
   }
 
+  /// Write the SkSL shaders to a zip file in build directory.
+  Future<void> writeSkSL() async {
+    if (!supportsWriteSkSL) {
+      throw Exception('writeSkSL is not supported by this runner.');
+    }
+    final Map<String, Object> data = await flutterDevices.first.views.first.getSkSLs();
+    if (data.isEmpty) {
+      globals.logger.printStatus(
+        'No data was receieved. To ensure SkSL data can be generated use a '
+        'physical device then:\n'
+        '  1. Pass "--cache-sksl" as an argument to flutter run.\n'
+        '  2. Interact with the application to force shaders to be compiled.\n'
+      );
+      return;
+    }
+    final File outputFile = globals.fsUtils.getUniqueFile(
+      globals.fs.currentDirectory,
+      'flutter',
+      'sksl',
+    );
+    final Map<String, Object> manifest = <String, Object>{
+      'device': getNameForTargetPlatform(await flutterDevices.first.device.targetPlatform),
+      'version': globals.flutterVersion.engineRevision,
```

Comment:
This version should be the git hash for the engine repo, right? It might be nice to rename version to engine_revision to make that more clear.

Reply:
Done